pkg/controllers/nodeclaim/podevents/suite_test.go

```diff
@@ -0,0 +1,169 @@
+			return []string{obj.(*v1.NodeClaim).Status.ProviderID}
+		})
+	}))
+	ctx = options.ToContext(ctx, test.Options())
+	cp = fake.NewCloudProvider()
+	podEventsController = podevents.NewController(fakeClock, env.Client)
+})
+
+var _ = AfterSuite(func() {
+	Expect(env.Stop()).To(Succeed(), "Failed to stop environment")
+})
+
+var _ = BeforeEach(func() {
+	ctx = options.ToContext(ctx, test.Options())
+	fakeClock.SetTime(time.Now())
+})
+
+var _ = AfterEach(func() {
+	cp.Reset()
+	ExpectCleanedUp(ctx, env.Client)
+})
+var _ = Describe("PodEvents", func() {
+	var nodePool *v1.NodePool
+	var nodeClaim *v1.NodeClaim
+	var node *corev1.Node
+	var pod *corev1.Pod
+
+	BeforeEach(func() {
+		nodePool = test.NodePool()
+		nodeClaim, node = test.NodeClaimAndNode(v1.NodeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					v1.NodePoolLabelKey:            nodePool.Name,
+					corev1.LabelInstanceTypeStable: "default-instance-type", // need the instance type for the cluster state update
+				},
+			},
+			Status: v1.NodeClaimStatus{
+				ProviderID: test.RandomProviderID(),
+			},
+		})
+		pod = test.Pod(test.PodOptions{
+			NodeName: node.Name,
+		})
+	})
+	It("should set the nodeclaim lastPodEvent", func() {
+		ExpectApplied(ctx, env.Client, nodePool, nodeClaim, node, pod)
+		fakeClock.SetTime(pod.CreationTimestamp.Time)
+		ExpectObjectReconciled(ctx, env.Client, podEventsController, pod)
+
+		nodeClaim = ExpectExists(ctx, env.Client, nodeClaim)
+		Expect(nodeClaim.Status.LastPodEventTime.Time).To(BeEquivalentTo(pod.CreationTimestamp.Time))
+	})
+	It("should not set the nodeclaim lastPodEvent for a node that Karpenter doesn't own", func() {
+		delete(node.Labels, v1.NodePoolLabelKey)
+		ExpectApplied(ctx, env.Client, nodePool, node, pod)
+		fakeClock.SetTime(pod.CreationTimestamp.Time)
+		ExpectObjectReconciled(ctx, env.Client, podEventsController, pod)
+
+		Expect(nodeClaim.Status.LastPodEventTime.Time).To(BeZero())
+	})
+	It("should not set the nodeclaim lastPodEvent when the node does not exist", func() {
+		ExpectApplied(ctx, env.Client, nodePool, nodeClaim, pod)
+		fakeClock.SetTime(pod.CreationTimestamp.Time)
+		_ = ExpectObjectReconcileFailed(ctx, env.Client, podEventsController, pod)
+
+		nodeClaim = ExpectExists(ctx, env.Client, nodeClaim)
+		Expect(nodeClaim.Status.LastPodEventTime.Time).To(BeZero())
+	})
+	It("should not set the nodeclaim lastPodEvent when the nodeclaim does not exist", func() {
+		ExpectApplied(ctx, env.Client, nodePool, node, pod)
+		fakeClock.SetTime(pod.CreationTimestamp.Time)
+		_ = ExpectObjectReconcileFailed(ctx, env.Client, podEventsController, pod)
+		Expect(nodeClaim.Status.LastPodEventTime.Time).To(BeZero())
+	})
+	It("should only set the nodeclaim lastPodEvent when it hasn't been set before", func() {
+		ExpectApplied(ctx, env.Client, nodePool, node, nodeClaim, pod)
+		fakeClock.SetTime(pod.CreationTimestamp.Time)
```

Comment:
Does it matter if this is set to the pod creation timestamp? It may be a little misleading IMO since this is really just ensuring that the lastPodEventTime should be set equal to clock.Now() (whatever that is)

Reply:
You're right this isn't needed